app/pages/activity/components/recentActivities.tsx

```diff
@@ -78,50 +82,52 @@ const RecentActivities: React.FC<RecentActivitiesProps> = ({
               <span>{log.title}</span>
               <span className="text-muted-foreground">{log.duration}</span>
             </div>
-            <div className="flex items-center space-x-2">
-              <Popover>
-                <PopoverTrigger>
-                  <Button>Log Progress</Button>
-                </PopoverTrigger>
-                <PopoverContent className="bg-card p-4 rounded-lg shadow-lg max-w-xs mt-4">
-                  <div className="flex flex-col items-start space-y-2">
-                    <Input
-                      type="number"
-                      value={progressInputs[log.id] || 0}
-                      onChange={(e) =>
-                        handleProgressChange(log.id, Number(e.target.value))
-                      }
-                    />
-                    <div className="flex space-x-2">
-                      <Button onClick={() => handleAdd(log.id)}>Add</Button>
-                      <Button onClick={() => handleSubtract(log.id)}>
-                        Subtract
-                      </Button>
-                      <Button onClick={() => handleLogProgress(log.id)}>
-                        Log
-                      </Button>
+            {hasAccess(Number(userId), loggedUser) && (
```

Comment:
you can make it much cooler tbh

create hook 
const {canModifyActivities}  = useUsersAccess(userId)
and inside hook return 

```
const loggedUser = // get logged user from react context , no prop drilling please
return { 
canModifyActivities: isSameUser(loggedUser, userId)
... all other activities
}
```

So in future we can stare permissions from one place